### PR TITLE
Partially fixed #148: search is not undefined anymore for few apps

### LINKED ISSUE #148: Add data toggle validator and respective data errors for all the apps
- Issue type: Bug Report

### Short description
Most of the apps in apps.loklak.org is supporting blank searches and showing results for undefined.

### Expected Behaviour
So here we can add a validator and display respective error messages when clicked on search button or when one of the field is skipped.

### PR DESCRIPTION
**Pull request by kavithaenair**
### Short description
Partially solves #148. This PR do not allow the below mentioned apps to do empty search.
The apps corrected in this PR and their test links:
LOKLAK TWEETS HISTOGRAM ABOUT FOSSASIA: https://kavithaenair.github.io/fossasia-histogram/ 
LOKLAK SEARCH PAGE: https://kavithaenair.github.io/yasoob/
SENTIMENT VISUALIZER: https://kavithaenair.github.io/sentimentVisualizer/
USER PROFILE PAGE: https://kavithaenair.github.io/userprofile/
LOKLAK SEARCH PAGE:https://kavithaenair.github.io/yathannsh/
**Enable the scripts to see the output on the browser**
In the next PR i will be adding few other apps and also the popup of error message.

I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

### CODE REVIEW

**Comment by kavithaenair** [comment]
@djmgit @SKrPl @vibhcool @daminisatya @mariobehling open for review.

File: histogram/index.html [hunk 1]
```diff
@@ -13,14 +13,15 @@
...
-    <a href="#" class='submit' ng-click="doSearch()">Plot</a>
+<form name="myform">
+    <label for="query" class="query">Query: </label><input ng-model="query" id="query"/ ng-required="true"><br>
+    <label for="from" class="from">From: </label><input ng-model="from" id="from" type="date" ng-required="true"/><br>
+    <label for="to" class="to">To: </label><input ng-model="to" id="to" type="date" ng-required="true"/><br>
+    <a href="#" class='submit' ng-click="myform.$valid && doSearch()">Plot</a>
 </form>
 <div id="bar-example" style="height: 250px;">
 </div>
+<script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.2.1/angular.min.js"></script>
```
Comment (djmgit) [inline histogram/index.html]: @kavithaenair Any reason for adding this here, as angular.min.js has already been included once above.

File: sentimentVisualizer/visualize.html [hunk 2]
```diff
@@ -19,6 +19,7 @@
         <script src="../../js/angular.min.js"></script>
         <script src="js/highcharts.min.js"></script>
         <script src="js/exporting.min.js"></script>
+        <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.2.1/angular.min.js"></script>
```
Comment (djmgit) [inline sentimentVisualizer/visualize.html]: here

File: histogram/index.html (hunk 1 above)
Comment (Achint08) [inline histogram/index.html]: @kavithaenair No use of pointing to CDN as angular.min.js is already installed locally. Remove it.

Comment (kavithaenair) [inline histogram/index.html]: @djmgit this is a different js version

File: sentimentVisualizer/visualize.html (hunk 2 above)
Comment (kavithaenair) [inline sentimentVisualizer/visualize.html]: Yup same here as well. The validation was working with this script link.

File: histogram/index.html (hunk 1 above)
Comment (Achint08) [inline histogram/index.html]: Moreover, try to add scripts like angular in the head so that they are loaded before the html is loaded.

**Comment by kavithaenair** [comment]
@djmgit thanks for the review. Will update the PR soon 😄

**Comment by kavithaenair** [comment]
@djmgit @Achint08 Please re-review.

**Comment by djmgit** [comment]
@kavithaenair 
searchapp

**Comment by kavithaenair** [comment]
@djmgit Sorry for the wrong information. Initially i haven't changed any files of LokLak search page app but mentioned in the PR description (by mistake).
Now updated the PR and the app is working fine. 👍 😄

**Pull request merged**